File: LeetcodeQuestions/leetcode_challenges/March_challenge/set_mismatch.rb
def find_error_nums(nums)
    arr = Array.new(nums.length).fill(0)
    res = Array.new(2)
    
    nums.each {|num| arr[num - 1] += 1}
    
    (0...arr.length).each do |i|
        if arr[i] == 2
            res[0] = i + 1
        end
        
        if arr[i] == 0
            res[1] = i + 1
        end
    end
    
    res
end

p find_error_nums([1,2,2,4])
p find_error_nums([1,1])

File: LeetcodeQuestions/leetcode_challenges/March_challenge/setMismatch.js
var findErrorNums = function (nums) {
    let arr = new Array(nums.length).fill(0)
    let res = new Array(2)

    for (let num of nums) {
        arr[num - 1] += 1
    }

    for (let i = 0; i < arr.length; i++) {
        if (arr[i] === 2) res[0] = i + 1
        if (arr[i] === 0) res[1] = i + 1
    }

    return res
};